cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/OperationHttpHandler.java

```diff
@@ -16,54 +16,112 @@
 
 package io.cdap.cdap.gateway.handlers;
 
+import com.google.common.collect.ImmutableMap;
 import com.google.gson.Gson;
+import com.google.inject.Inject;
+import io.cdap.cdap.api.feature.FeatureFlagsProvider;
+import io.cdap.cdap.common.BadRequestException;
+import io.cdap.cdap.common.ForbiddenException;
+import io.cdap.cdap.common.conf.CConfiguration;
 import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.common.conf.Constants.AppFabric;
+import io.cdap.cdap.common.feature.DefaultFeatureFlagsProvider;
+import io.cdap.cdap.features.Feature;
 import io.cdap.cdap.gateway.handlers.util.AbstractAppFabricHttpHandler;
+import io.cdap.cdap.internal.operation.OperationLifecycleManager;
+import io.cdap.cdap.internal.operation.OperationRunFilter;
+import io.cdap.cdap.internal.operation.OperationRunNotFoundException;
+import io.cdap.cdap.internal.operation.ScanOperationRunsRequest;
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.cdap.proto.id.OperationRunId;
 import io.cdap.cdap.proto.operation.OperationRun;
+import io.cdap.cdap.proto.operation.OperationRunStatus;
+import io.cdap.cdap.proto.operation.OperationType;
 import io.cdap.http.HttpHandler;
 import io.cdap.http.HttpResponder;
 import io.netty.handler.codec.http.FullHttpRequest;
 import io.netty.handler.codec.http.HttpRequest;
 import io.netty.handler.codec.http.HttpResponseStatus;
-import java.util.ArrayList;
-import java.util.List;
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import javax.ws.rs.GET;
 import javax.ws.rs.POST;
 import javax.ws.rs.Path;
 import javax.ws.rs.PathParam;
 import javax.ws.rs.QueryParam;
 
-/**
- * The {@link HttpHandler} for handling REST calls to operation endpoints.
- */
+/** The {@link HttpHandler} for handling REST calls to operation endpoints. */
 @Path(Constants.Gateway.API_VERSION_3 + "/namespaces/{namespace-id}/operations")
 public class OperationHttpHandler extends AbstractAppFabricHttpHandler {
-
+  private final CConfiguration cConf;
+  private static final Pattern KEY_VALUE_PATTERN = Pattern.compile("(\"?)(\\w+)=(\\w+)(\"?)");
+  private static final String FILTER_SPLITTER = "AND";
+  private final FeatureFlagsProvider featureFlagsProvider;
   private static final Gson GSON = new Gson();
-  
-  OperationHttpHandler() {
+  private final OperationLifecycleManager operationLifecycleManager;
+  private final int batchSize;
+  public static final String OPERATIONS_LIST_PAGINATED_KEY = "operations";
+
+  @Inject
+  OperationHttpHandler(CConfiguration cConf, OperationLifecycleManager operationLifecycleManager)
+      throws Exception {
+    this.cConf = cConf;
```

Comment:
remove field `cConf` as it is only used inside constructor

Reply:
done